src/units.cpp

```diff
@@ -594,8 +602,8 @@ bool Units::compatible(const UnitsPtr &units1, const UnitsPtr &units2)
 
 bool Units::equivalent(const UnitsPtr &units1, const UnitsPtr &units2)
 {
-    // Units must be compatible to return a non-zero scaling factor.
-    return Units::scalingFactor(units1, units2) == 1.0;
+    // Units must be compatible and return a scaling factor of 1.0.
+    return Units::scalingFactor(units1, units2, true) == 1.0;
```

Comment:
Why pass `true` to `Units::scalingFactor()` since it's the default value for that parameter... or am I missing something?...

Reply:
Oops, I thought it was false as a default? There was a test failing until I added this (though if you're right then it was for other reasons ... !). Either way the comment needed updating :). Feel free to remove the default.

Reply:
Done.